community/cypher/cypher-compiler-2.3/src/test/scala/org/neo4j/cypher/internal/compiler/v2_3/test_helpers/RandomizedTestSupport.scala

```diff
@@ -20,6 +20,16 @@
 package org.neo4j.cypher.internal.compiler.v2_3.test_helpers
 
 trait RandomizedTestSupport {
-  val N: Double  = testRandomizer.nextDouble() * 1E6
+
+  /**
+   * Minimum number of nodes in the database.
+   * This value is needed for correct cardinality estimations.
+   */
+  private val MIN_N = 1500
```

Comment:
For small number of nodes chosen plan does not matter, so does the estimated cardinality.
At the same time small number of `N` cause excessive test flakiness because of rounding errors.